src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -436,6 +437,21 @@ module.exports = function<T, P, I, TI, C, CX>(
     } else {
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
+      if (__DEV__) {
+        let info = '';
+        const ownerName = ReactDebugCurrentFiber.getCurrentFiberOwnerName();
+        if (ownerName) {
+          info += ' Check the render method of `' + ownerName + '`.';
+        }
+
+        warning(
+          !workInProgress.ref,
```

Comment:
One unfortunate thing related to our `warning` transformation is that `getCurrentFiberOwnerName()` runs regardless of this condition. Since this is a hot path in dev, can you please hide it behind a check?

```js
if (workInProgress.ref != null) {
  ...
  warning(false, ...)
}
```

Yes, this code is funny.